app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+            $this->info("Unable to find 'git' on the system so I cannot initialize a git repo in '{$this->projectpath}'");
+
+            return false;
+        }
+
+        $process = new Process('dummy command');
+        $process->setWorkingDirectory($this->projectpath);
+
+        $commands = [
+            'git init',
+            'git add .',
+            'git commit -m "'.str_replace('"', '\"', $this->commitmessage).'"',
+        ];
+
+        foreach ($commands as $command) {
+            $process->setCommandLine($command);
+            $process->run();
+            $this->line($process->getOutput());
+        }
+
+        return true;
+    }
+
+    protected function openTextEditor()
+    {
+        if ($this->option('editor')) {
+            $editor = $this->option('editor');
+        } else {
+            $finder = new ExecutableFinder();
+            foreach ($this->editors_gui as $_editor) {
+                if ($finder->find($_editor)) {
+                    $editor = $_editor;
+                    break;
+                }
+            }
+            if (empty($editor)) {
+                foreach ($this->editors_terminal as $_editor) {
+                    if ($finder->find($_editor)) {
+                        $editor = $_editor;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (empty($editor)) {
+            $this->warn('Unable to find a text editor to open, skipping this step.');
+
+            return false;
+        }
+
+        $this->info("Found editor $editor so an opening it now");
+
+        $process = new Process("$editor .");
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->setTty(true);
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        return true;
+    }
+
+    protected function createDatabase($type = 'sqlite')
+    {
+        if ($type === 'sqlite') {
+            $basedir = $this->projectpath.DIRECTORY_SEPARATOR.'database';
+
+            // @todo Could not find nice way to add (touch) an empty file here with Filesystem so am doing it manually
+            $process = new Process('touch database.sqlite');
+            $process->setWorkingDirectory($basedir);
+            $process->run();
+
+            $this->info($process->getOutput());
+
+            $this->replaceEnvVariable('DB_CONNECTION', 'sqlite');
+            // @todo Seems this must be a full path, true? Also, should we use a config/ file instead of .env?
+            $this->replaceEnvVariable('DB_DATABASE', $this->projectpath.'/database/database.sqlite');
+
+            return true;
+        }
+
+        if ($type === 'mysql') {
+            // @todo Offer ways to create users here, one user per db? Auto-generate passwords or always use the same?
+
+            // @todo sanitize this name
+            $sql = "CREATE database {$this->projectname}";
```

Comment:
I think this will work?

```php
DB::getConnection()->statement('CREATE DATABASE :dbname', ['dbname' => $this->projectname]);

```

Reply:
I spent time on this but couldn't get it to work. The challenge here is we're essentially defining a dynamic connection for either sqlite or mysql and using the user's custom configuration options (lambo doesn't handle that yet -- default is root with empty pass) all with a connection without a database (most Laravel connection logic assumes a database exists and is used). I've also run into errors that I don't yet understand so would prefer postponing this part if that's okay. But, on the bright side, the sqlite part works :)

Reply:
OK, cool! I'll toy around with it once we get this merged--I have some ideas. :)

Reply:
👍